src/analysis/convex/basic.lean

```diff
@@ -625,85 +671,217 @@ begin
+begin
+  refine subset.antisymm (convex_hull_min _ _) _,
+  { intros x hx,
+    use [punit, finset.singleton punit.star, λ _, 1, λ _, x, λ _ _, zero_le_one,
+      finset.sum_singleton, λ _ _, hx],
+    simp only [finset.center_mass, finset.sum_singleton, inv_one, one_smul] },
+  { rintros x y ⟨α, sx, wx, zx, hwx₀, hwx₁, hzx, rfl⟩ ⟨β, sy, wy, zy, hwy₀, hwy₁, hzy, rfl⟩
+      a b ha hb hab,
+    rw [finset.center_mass_segment' _ _ _ _ _ _ hwx₁ hwy₁ _ _ hab],
+    refine ⟨_, _, _, _, _, _, _, rfl⟩,
+    { rintros i hi,
+      rw [finset.mem_union, finset.mem_image, finset.mem_image] at hi,
+      rcases hi with ⟨j, hj, rfl⟩|⟨j, hj, rfl⟩;
+        simp only [sum.elim_inl, sum.elim_inr];
+        apply_rules [mul_nonneg, hwx₀, hwy₀] },
+    { simp [finset.sum_sum_elim, finset.mul_sum.symm, *] },
+    { intros i hi,
+      rw [finset.mem_union, finset.mem_image, finset.mem_image] at hi,
+      rcases hi with ⟨j, hj, rfl⟩|⟨j, hj, rfl⟩;
+        simp only [sum.elim_inl, sum.elim_inr]; apply_rules [hzx, hzy] } },
+  { rintros _ ⟨α, t, w, z, hw₀, hw₁, hz, rfl⟩,
+    exact t.center_mass_mem_convex_hull hw₀ (hw₁.symm ▸ zero_lt_one) hz }
+end
+
+lemma set.finite.convex_hull_eq {s : set E} (hs : finite s) :
+  convex_hull s = {x : E | ∃ (w : E → ℝ) (hw₀ : ∀ y ∈ s, 0 ≤ w y) (hw₁ : hs.to_finset.sum w = 1),
+    hs.to_finset.center_mass w id = x} :=
+begin
+  refine subset.antisymm (convex_hull_min _ _) _,
+  { intros x hx,
+    replace hx : x ∈ hs.to_finset, from finite.mem_to_finset.2 hx,
+    refine ⟨_, _, _, finset.center_mass_ite_eq _ _ _ hx⟩,
+    { intros, split_ifs, exacts [zero_le_one, le_refl 0] },
+    { rw [finset.sum_ite_eq, if_pos hx] } },
+  { rintros x y ⟨wx, hwx₀, hwx₁, rfl⟩ ⟨wy, hwy₀, hwy₁, rfl⟩
+      a b ha hb hab,
+    rw [finset.center_mass_segment _ _ _ _ hwx₁ hwy₁ _ _ hab],
+    refine ⟨_, _, _, rfl⟩,
+    { rintros i hi,
+      apply_rules [add_nonneg, mul_nonneg, hwx₀, hwy₀], },
+    { simp only [finset.sum_add_distrib, finset.mul_sum.symm, mul_one, *] } },
+  { rintros _ ⟨w, hw₀, hw₁, rfl⟩,
+    exact hs.to_finset.center_mass_mem_convex_hull (λ x hx, hw₀ _ $ finite.mem_to_finset.1 hx)
+      (hw₁.symm ▸ zero_lt_one) (λ x hx, finite.mem_to_finset.1 hx) }
+end
+
+lemma is_linear_map.convex_hull_image {f : E → F} (hf : is_linear_map ℝ f) (s : set E) :
+  convex_hull (f '' s) = f '' convex_hull s :=
+set.subset.antisymm (convex_hull_min (image_subset _ (subset_convex_hull s)) $
+  (convex_convex_hull s).is_linear_image hf)
+  (image_subset_iff.2 $ convex_hull_min
+    (image_subset_iff.1 $ subset_convex_hull _)
+    ((convex_convex_hull _).is_linear_preimage hf))
+
+lemma linear_map.convex_hull_image (f : E →ₗ[ℝ] F) (s : set E) :
+  convex_hull (f '' s) = f '' convex_hull s :=
+f.is_linear.convex_hull_image s
+
+end convex_hull
+
+/-! ### Simplex -/
+
+section simplex
+
+variables (α) [fintype α] {f : α → ℝ}
+
+/-- Standard simplex in the space of functions `α → ℝ`. -/
```

Comment:
Recall in the docstring what this means.

Reply:
Done.